documentation/modules/oauth/con-oauth-reauthentication.adoc

```diff
@@ -1,23 +1,39 @@
+This Kafka mechanism is used to enforce the expiry of authenticated sessions between OAuth clients and Kafka brokers after a defined period of time. 
+When a session expires, the client immediately starts a new session by reusing the existing connection rather than dropping it.
 
-When the re-authentication mechanism is enabled in the configuration of an `oauth` type listener, the broker's authenticated session expires when the access token expires.
-The client must then re-authenticate to the existing session by sending a new, valid access token to the broker, without dropping the connection.
+Session re-authentication is disabled by default. 
+To enable and configure it, you set a time value for `maxSecondsWithoutReauthentication` in the `oauth` listener configuration in the `Kafka` custom resource. 
+The same property is used to configure session re-authentication for OAUTHBEARER and PLAIN authentication. 
+For an example configuration, see xref:proc-oauth-authentication-broker-config-{context}[].
+
+Session re-authentication can be used with _fast local JWT_ or _introspection endpoint_ token validation.
+
+.Client re-authentication
+
+When the broker's authenticated session expires, the client must re-authenticate to the existing session by sending a new, valid access token to the broker, without dropping the connection.
 
 If token validation is successful, a new client session is started using the existing connection.
 If the client fails to re-authenticate, the broker will close the connection if further attempts are made to send or receive messages.
 Java clients that use Kafka client library 2.2 or later automatically re-authenticate if the re-authentication mechanism is enabled on the broker.
 
-You enable session re-authentication in the `Kafka` resource. Set the `maxSecondsWithoutReauthentication` property for a TLS listener with `type: oauth` authentication.
-Session re-authentication is supported for both types of token validation (fast local JWT and introspection endpoint). For an example configuration, see xref:proc-oauth-authentication-broker-config-{context}[].
+.Differences in session expiry for OAUTHBEARER and PLAIN
 
-For more information about the re-authentication mechanism, which was added in Kafka version 2.2, see link:https://cwiki.apache.org/confluence/display/KAFKA/KIP-368%3A+Allow+SASL+Connections+to+Periodically+Re-Authenticate[KIP-368^].
+There are some differences in how session expiry works when session re-authentication is configured for OAUTHBEARER and PLAIN authentication.
+
+* For OAUTHBEARER, the broker's authenticated session will expire at the configured `maxSecondsWithoutReauthentication`. 
+The session will expire earlier if the access token expires before the `maxSecondsWithoutReauthentication`. 
+The duration of the access token always take precedence over `maxSecondsWithoutReauthentication`.
```

Comment:
This last statement seems to me to sow confusion.
First we say session will expire at the configured time unless the token expires beforehand in which case the session will expire earlier.
Then we say token expiry overrides the configured time - meaning if token lifetime is longer it also overrides the configured time.

If `maxSecondsWithoutReauthentication` is not configured, the session will not expire with the expiry of the access token. This is the same as for PLAIN - without `maxSecondsWithoutReauthentication` the sessions just don't expire by themselves. 

```suggestion
```